packages/ui-lib/src/Misc/molecules/Pagination.tsx

```diff
@@ -0,0 +1,351 @@
+import React, { ChangeEvent, HTMLAttributes, useCallback, useState, useRef, Fragment } from 'react';
+import styled, { keyframes } from 'styled-components';
+import Button from '../../Form/atoms/Button';
+import Icon from '../../Icons/Icon';
+import { removeAutoFillStyle, resetButtonStyles } from '../../common';
+import { isNotNumber } from '../../helpers';
+import SelectField, { SelectWrapper } from '../../Form/atoms/SelectField';
+
+const WIDTH_PER_NUMBER = 12;
+
+const PaginationContainer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  padding-top: 20px;
+  padding-bottom: 20px;
+  width: 100%;
+  height: fit-content;
+  margin-right: 10px;
+  white-space: nowrap;
+  gap: 0 8px;
+  vertical-align: baseline;
+`;
+
+const PageLabel = styled.label`
+  font-family: var(--font-ui);
+  font-weight: 500px;
+  font-size: 14px;
+  color: var(--grey-10);
+  text-align: left;
+`;
+
+const StaticPageCount = styled.div`
+  display: flex;
+  align-items: center;
+  box-sizing: border-box;
+  font-size: 14px;
+  color: var(--grey-8);
+  height: 100%;
+  text-align: left;
+  padding-top: 1px;
+  padding-right: 1px;
+`;
+
+const StyledInput = styled.input<{ textColor: string, maxWidth?: string }>`
+  ${removeAutoFillStyle};
+  color: ${({ textColor }) => textColor};
+  max-width: ${({ maxWidth }) => maxWidth ? maxWidth : '40px'};
+  font-family: var(--font-data);
+  height: 100%;
+  box-sizing: border-box;
+  outline: none;
+  background: transparent;
+  text-align: right;
+  font-size: 14px;
+  font-weight: 500;
+  line-height: 25px;
+  border: none;
+  padding-right: ${WIDTH_PER_NUMBER / 2}px;
+`;
+
+const shakeAnimation = keyframes`
+  0% { transform: translateX(0); }
+  10%, 30%, 50%, 70%, 90% { transform: translateX(-5px); }
+  20%, 40%, 60%, 80% { transform: translateX(5px); }
+  100% { transform: translateX(0); }
+`;
+
+const InputContainer = styled.div<{ borderColor?: string, shouldShake: boolean }>`
+  height: var(--input-height, 40px);
+  animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
+  -moz-animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
+  -webkit-animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
+  -o-animation: ${({ shouldShake }) => (shouldShake ? shakeAnimation : 'none')} 150ms 2 linear;
```

Comment:
I think we can drop browser prefixes here. I'm pretty sure compiler adds these for properties that still require it.